Benchmarks are not working properly

hi,
My ubuntu version is below
```
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.4 LTS
Release:        18.04
Codename:       bionic
```
How to Reproduce the problem
refer to this document：https://github.com/cosmos/iavl/blob/master/benchmarks/README.md
1. I clone  iavl to my machine.
```
git clone https://github.com/cosmos/iavl.git
```
2. go to the setup position
```
cd iavl/benchmarks/setup/
```
3. run the INSTALL_ROOT.sh
```
./INSTALL_ROOT.sh
```
4. run the benchmarks
```
./RUN_BENCHMARKS.sh
```
then I go the error:
```
Cloning into 'iavl'...
remote: Enumerating objects: 6241, done.
remote: Counting objects: 100% (682/682), done.
remote: Compressing objects: 100% (418/418), done.
remote: Total 6241 (delta 433), reused 450 (delta 260), pack-reused 5559
Receiving objects: 100% (6241/6241), 9.43 MiB | 15.96 MiB/s, done.
Resolving deltas: 100% (4062/4062), done.
Already on 'master'
Your branch is up to date with 'origin/master'.
make: *** [bench] Error 1
```
5. I go to the iavl position
```
cd ~/iavl/
```
6. run benchmark
```
make bench
```
then I got the result below:
```
cd benchmarks && \
	go test -ldflags "-X github.com/cosmos/iavl.Version=0.18.0-17-g80bc6e4 -X github.com/cosmos/iavl.Commit=80bc6e464bbbaac360f27ddc570f4511535cfec6 -X github.com/cosmos/iavl.Branch=master" -bench=RandomBytes . && \
	go test -ldflags "-X github.com/cosmos/iavl.Version=0.18.0-17-g80bc6e4 -X github.com/cosmos/iavl.Commit=80bc6e464bbbaac360f27ddc570f4511535cfec6 -X github.com/cosmos/iavl.Branch=master" -tags cleveldb,rocksdb,boltdb,badgerdb -bench=Small . && \
	go test -ldflags "-X github.com/cosmos/iavl.Version=0.18.0-17-g80bc6e4 -X github.com/cosmos/iavl.Commit=80bc6e464bbbaac360f27ddc570f4511535cfec6 -X github.com/cosmos/iavl.Branch=master" -tags cleveldb,rocksdb,boltdb,badgerdb -bench=Medium . && \
	go test -ldflags "-X github.com/cosmos/iavl.Version=0.18.0-17-g80bc6e4 -X github.com/cosmos/iavl.Commit=80bc6e464bbbaac360f27ddc570f4511535cfec6 -X github.com/cosmos/iavl.Branch=master" -bench=BenchmarkMemKeySizes .
iavl: 0.18.0-17-g80bc6e4
git commit: 80bc6e464bbbaac360f27ddc570f4511535cfec6
git branch: master
go version go1.17.6 linux/amd64

goos: linux
goarch: amd64
pkg: github.com/cosmos/iavl/benchmarks
cpu: Intel(R) Xeon(R) Platinum 8375C CPU @ 2.90GHz
BenchmarkRandomBytes/random-4-16         	46490554	        24.76 ns/op
BenchmarkRandomBytes/random-16-16        	31667058	        37.41 ns/op
BenchmarkRandomBytes/random-32-16        	23234440	        51.63 ns/op
BenchmarkRandomBytes/random-100-16       	10534645	       113.5 ns/op
BenchmarkRandomBytes/random-1000-16      	 1285242	       936.5 ns/op
PASS
ok  	github.com/cosmos/iavl/benchmarks	7.125s
iavl: 0.18.0-17-g80bc6e4
git commit: 80bc6e464bbbaac360f27ddc570f4511535cfec6
git branch: master
go version go1.17.6 linux/amd64

Init Tree took 1.11 MB
goos: linux
goarch: amd64
pkg: github.com/cosmos/iavl/benchmarks
cpu: Intel(R) Xeon(R) Platinum 8375C CPU @ 2.90GHz
BenchmarkSmall/memdb-1000-100-4-10/query-no-in-tree-guarantee-fast-16         	 3373084	       354.2 ns/op	      64 B/op	       3 allocs/op
--- FAIL: BenchmarkSmall/memdb-1000-100-4-10/query-no-in-tree-guarantee-slow
    bench_test.go:93:
        	Error Trace:	bench_test.go:93
        	            				bench_test.go:381
        	            				benchmark.go:192
        	            				benchmark.go:232
        	            				asm_amd64.s:1581
        	Error:      	Should be false
        	Test:       	BenchmarkSmall/memdb-1000-100-4-10/query-no-in-tree-guarantee-slow
BenchmarkSmall/memdb-1000-100-4-10/query-hits-fast-16                         	15130903	        79.20 ns/op	       0 B/op	       0 allocs/op
--- FAIL: BenchmarkSmall/memdb-1000-100-4-10/query-hits-slow
    bench_test.go:113:
        	Error Trace:	bench_test.go:113
        	            				bench_test.go:390
        	            				benchmark.go:192
        	            				benchmark.go:232
        	            				asm_amd64.s:1581
        	Error:      	Should be false
        	Test:       	BenchmarkSmall/memdb-1000-100-4-10/query-hits-slow
BenchmarkSmall/memdb-1000-100-4-10/iteration-fast-16                          	    4292	    274672 ns/op	  152498 B/op	    2999 allocs/op
--- BENCH: BenchmarkSmall/memdb-1000-100-4-10/iteration-fast-16
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
	... [output truncated]
BenchmarkSmall/memdb-1000-100-4-10/iteration-slow-16                          	     735	   1626761 ns/op	  819175 B/op	   15796 allocs/op
--- BENCH: BenchmarkSmall/memdb-1000-100-4-10/iteration-slow-16
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
	... [output truncated]
BenchmarkSmall/memdb-1000-100-4-10/update-16                                  	   51558	     25466 ns/op	   12071 B/op	     183 allocs/op
BenchmarkSmall/memdb-1000-100-4-10/block-16                                   	     314	   5650135 ns/op	 2367334 B/op	   36232 allocs/op
--- FAIL: BenchmarkSmall/memdb-1000-100-4-10
Init Tree took 0.72 MB
BenchmarkSmall/goleveldb-1000-100-4-10/query-no-in-tree-guarantee-fast-16     	 1589504	       755.9 ns/op	     120 B/op	       6 allocs/op
--- FAIL: BenchmarkSmall/goleveldb-1000-100-4-10/query-no-in-tree-guarantee-slow
    bench_test.go:93:
        	Error Trace:	bench_test.go:93
        	            				bench_test.go:381
        	            				benchmark.go:192
        	            				benchmark.go:232
        	            				asm_amd64.s:1581
        	Error:      	Should be false
        	Test:       	BenchmarkSmall/goleveldb-1000-100-4-10/query-no-in-tree-guarantee-slow
BenchmarkSmall/goleveldb-1000-100-4-10/query-hits-fast-16                     	15140086	        79.31 ns/op	       0 B/op	       0 allocs/op
--- FAIL: BenchmarkSmall/goleveldb-1000-100-4-10/query-hits-slow
    bench_test.go:113:
        	Error Trace:	bench_test.go:113
        	            				bench_test.go:390
        	            				benchmark.go:192
        	            				benchmark.go:232
        	            				asm_amd64.s:1581
        	Error:      	Should be false
        	Test:       	BenchmarkSmall/goleveldb-1000-100-4-10/query-hits-slow
BenchmarkSmall/goleveldb-1000-100-4-10/iteration-fast-16                      	    2059	    589798 ns/op	  168480 B/op	    4997 allocs/op
--- BENCH: BenchmarkSmall/goleveldb-1000-100-4-10/iteration-fast-16
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
	... [output truncated]
BenchmarkSmall/goleveldb-1000-100-4-10/iteration-slow-16                      	     458	   2601788 ns/op	 1097276 B/op	   23739 allocs/op
--- BENCH: BenchmarkSmall/goleveldb-1000-100-4-10/iteration-slow-16
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
    bench_test.go:156: completed 1000 iterations
	... [output truncated]
BenchmarkSmall/goleveldb-1000-100-4-10/update-16                              	   33482	     38438 ns/op	   21474 B/op	     184 allocs/op
BenchmarkSmall/goleveldb-1000-100-4-10/block-16                               	     200	   8488533 ns/op	 4255632 B/op	   42457 allocs/op
--- FAIL: BenchmarkSmall/goleveldb-1000-100-4-10
--- FAIL: BenchmarkSmall
FAIL
exit status 1
FAIL	github.com/cosmos/iavl/benchmarks	20.050s
FAIL
Makefile:48: recipe for target 'bench' failed
make: *** [bench] Error 1
```
It seems error occurs on this position， the isFastCacheEnabled should be false:
```
	isFastCacheEnabled, err := t.IsFastCacheEnabled() // to ensure fast storage is enabled
	require.NoError(b, err)
	require.False(b, isFastCacheEnabled) // to ensure fast storage is not enabled
```
----------------------------------
Am I missing something？I expect the correct benchmark result. and how can I help?

Hi @p0mvn . When I use rocksdb. the batch panics.
```
Init Tree took 0.57 MB
BenchmarkSmall/rocksdb-1000-100-4-10/query-no-in-tree-guarantee-fast-16       	  951153	      1295 ns/op	      72 B/op	       7 allocs/op
BenchmarkSmall/rocksdb-1000-100-4-10/query-no-in-tree-guarantee-slow-16       	  428944	      3223 ns/op	     476 B/op	      14 allocs/op
BenchmarkSmall/rocksdb-1000-100-4-10/query-hits-fast-16                       	14625267	        82.67 ns/op	       0 B/op	       0 allocs/op
BenchmarkSmall/rocksdb-1000-100-4-10/query-hits-slow-16                       	fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x1 addr=0x7feff6734958 pc=0x7feff2235108]

runtime stack:
runtime.throw({0xb0ad87, 0x3a16178})
	/usr/local/go/src/runtime/panic.go:1198 +0x71
runtime.sigpanic()
	/usr/local/go/src/runtime/signal_unix.go:719 +0x396

goroutine 6503 [syscall]:
runtime.cgocall(0x9560c0, 0xc00006d940)
	/usr/local/go/src/runtime/cgocall.go:156 +0x5c fp=0xc00006d918 sp=0xc00006d8e0 pc=0x421dbc
github.com/tecbot/gorocksdb._Cfunc_rocksdb_create_iterator(0x38f8038, 0x396bde0)
	_cgo_gotypes.go:1162 +0x4d fp=0xc00006d940 sp=0xc00006d918 pc=0x86570d
github.com/tecbot/gorocksdb.(*DB).NewIterator.func1(0x93305f, 0xc0001de7e0)
	/root/go/pkg/mod/github.com/tecbot/gorocksdb@v0.0.0-20191217155057-f0fad39f321c/db.go:499 +0x7d fp=0xc00006d980 sp=0xc00006d940 pc=0x87785d
github.com/tecbot/gorocksdb.(*DB).NewIterator(0xc000050800, 0x0)
	/root/go/pkg/mod/github.com/tecbot/gorocksdb@v0.0.0-20191217155057-f0fad39f321c/db.go:499 +0x19 fp=0xc00006d9a8 sp=0xc00006d980 pc=0x877799
github.com/tendermint/tm-db.(*RocksDB).ReverseIterator(0xc0001de7e0, {0xc000922040, 0x1, 0x9}, {0xc000922050, 0x9, 0x9})
	/root/go/pkg/mod/github.com/tendermint/tm-db@v0.6.6/rocksdb.go:198 +0x85 fp=0xc00006da00 sp=0xc00006d9a8 pc=0x898125
github.com/cosmos/iavl.(*nodeDB).getPreviousVersion(0xc000fc4000, 0xc0019e6000)
	/root/zhangkai/iavl/nodedb.go:765 +0x130 fp=0xc00006dae0 sp=0xc00006da00 pc=0x93f0f0
github.com/cosmos/iavl.(*nodeDB).SaveOrphans(0xc000fc4000, 0x0, 0xa27600)
	/root/zhangkai/iavl/nodedb.go:658 +0xc5 fp=0xc00006dc28 sp=0xc00006dae0 pc=0x93e385
github.com/cosmos/iavl.(*MutableTree).SaveVersion(0xc000f1a000)
	/root/zhangkai/iavl/mutable_tree.go:861 +0x165 fp=0xc00006dde8 sp=0xc00006dc28 pc=0x935565
github.com/cosmos/iavl/benchmarks.runKnownQueriesSlow(0xc0001446c0, 0x4a34d7, {0xc000fcc000, 0x3e8, 0x1226640})
	/root/zhangkai/iavl/benchmarks/bench_test.go:106 +0x52 fp=0xc00006de88 sp=0xc00006dde8 pc=0x951732
github.com/cosmos/iavl/benchmarks.runSuite.func4(0xc0001446c0)
	/root/zhangkai/iavl/benchmarks/bench_test.go:390 +0x34 fp=0xc00006dec0 sp=0xc00006de88 pc=0x953fd4
testing.(*B).runN(0xc0001446c0, 0x64)
	/usr/local/go/src/testing/benchmark.go:192 +0x126 fp=0xc00006df40 sp=0xc00006dec0 pc=0x515a66
testing.(*B).launch(0xc0001446c0)
	/usr/local/go/src/testing/benchmark.go:328 +0x1c5 fp=0xc00006dfc8 sp=0xc00006df40 pc=0x516645
testing.(*B).doBench·dwrap·4()
	/usr/local/go/src/testing/benchmark.go:283 +0x26 fp=0xc00006dfe0 sp=0xc00006dfc8 pc=0x516446
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1581 +0x1 fp=0xc00006dfe8 sp=0xc00006dfe0 pc=0x486561
created by testing.(*B).doBench
	/usr/local/go/src/testing/benchmark.go:283 +0x7b

goroutine 1 [chan receive]:
testing.(*B).run1(0xc0000d66c0)
	/usr/local/go/src/testing/benchmark.go:234 +0xbb
testing.(*B).Run(0xc000144000, {0xaf55b6, 0x3d0b94b7fb362a}, 0xb1d6b0)
	/usr/local/go/src/testing/benchmark.go:670 +0x438
testing.runBenchmarks.func1(0xc000144000)
	/usr/local/go/src/testing/benchmark.go:544 +0x6e
testing.(*B).runN(0xc000144000, 0x1)
	/usr/local/go/src/testing/benchmark.go:192 +0x126
testing.runBenchmarks({0xb041ee, 0x21}, 0x0, {0x1199d00, 0x7, 0xaf5ce0})
	/usr/local/go/src/testing/benchmark.go:553 +0x3e6
testing.(*M).Run(0xc00011e080)
	/usr/local/go/src/testing/testing.go:1510 +0x6d6
main.main()
	_testmain.go:55 +0x14b

goroutine 24 [chan receive]:
testing.(*B).run1(0xc000144480)
	/usr/local/go/src/testing/benchmark.go:234 +0xbb
testing.(*B).Run(0xc0000d66c0, {0xc000026108, 0x0}, 0xc000fc2410)
	/usr/local/go/src/testing/benchmark.go:670 +0x438
github.com/cosmos/iavl/benchmarks.runBenchmarks(0xc0000d66c0, {0xc0000abeb0, 0x2, 0xb1f458})
	/root/zhangkai/iavl/benchmarks/bench_test.go:349 +0x67e
github.com/cosmos/iavl/benchmarks.BenchmarkSmall(0xc0000d66c0)
	/root/zhangkai/iavl/benchmarks/bench_test.go:281 +0xd7
testing.(*B).runN(0xc0000d66c0, 0x1)
	/usr/local/go/src/testing/benchmark.go:192 +0x126
testing.(*B).run1.func1()
	/usr/local/go/src/testing/benchmark.go:232 +0x59
created by testing.(*B).run1
	/usr/local/go/src/testing/benchmark.go:225 +0xa5

goroutine 6500 [chan receive]:
testing.(*B).doBench(0xc0001446c0)
	/usr/local/go/src/testing/benchmark.go:284 +0x8e
testing.(*benchContext).processBench(0xc000120048, 0x230)
	/usr/local/go/src/testing/benchmark.go:583 +0x3aa
testing.(*B).run(0xc0001446c0)
	/usr/local/go/src/testing/benchmark.go:275 +0x67
testing.(*B).Run(0xc000144480, {0xaf6677, 0xafc937}, 0xc00051c000)
	/usr/local/go/src/testing/benchmark.go:671 +0x446
github.com/cosmos/iavl/benchmarks.runSuite(0xc000144480, {0xbc4e90, 0xc003572120}, 0x3e8, 0x64, 0x4, 0xa)
	/root/zhangkai/iavl/benchmarks/bench_test.go:388 +0x30d
github.com/cosmos/iavl/benchmarks.runBenchmarks.func2(0xc000144480)
	/root/zhangkai/iavl/benchmarks/bench_test.go:350 +0x31
testing.(*B).runN(0xc000144480, 0x1)
	/usr/local/go/src/testing/benchmark.go:192 +0x126
testing.(*B).run1.func1()
	/usr/local/go/src/testing/benchmark.go:232 +0x59
created by testing.(*B).run1
	/usr/local/go/src/testing/benchmark.go:225 +0xa5
exit status 2
FAIL	github.com/cosmos/iavl/benchmarks	18.880s
FAIL
Makefile:48: recipe for target 'bench' failed
make: *** [bench] Error 1
```

--- a/benchmarks/bench_test.go
+++ b/benchmarks/bench_test.go
@@ -88,7 +88,7 @@ func runQueriesSlow(b *testing.B, t *iavl.MutableTree, keyLen int) {
 
 	itree, err := t.GetImmutable(version - 1)
 	require.NoError(b, err)
-	isFastCacheEnabled, err := t.IsFastCacheEnabled() // to ensure fast storage is enabled
+	isFastCacheEnabled, err := itree.IsFastCacheEnabled() // to ensure fast storage is enabled
 	require.NoError(b, err)
 	require.False(b, isFastCacheEnabled) // to ensure fast storage is not enabled
 
@@ -108,7 +108,7 @@ func runKnownQueriesSlow(b *testing.B, t *iavl.MutableTree, keys [][]byte) {
 
 	itree, err := t.GetImmutable(version - 1)
 	require.NoError(b, err)
-	isFastCacheEnabled, err := t.IsFastCacheEnabled() // to ensure fast storage is not enabled
+	isFastCacheEnabled, err := itree.IsFastCacheEnabled() // to ensure fast storage is not enabled
 	require.NoError(b, err)
 	require.False(b, isFastCacheEnabled)
 	b.StartTimer()

